config/src/genesis.rs

```diff
@@ -12,22 +16,85 @@ view! {
     pub struct Configuration {
         /// The public key of the genesis account, should be supplied to all peers.
         #[config(serde_as_str)]
-        pub account_public_key: PublicKey,
+        pub public_key: PublicKey,
         /// The private key of the genesis account, only needed for the peer that submits the genesis block.
         #[view(ignore)]
-        pub account_private_key: Option<PrivateKey>,
+        pub private_key: Option<PrivateKey>,
+        /// Path to the genesis file
+        #[config(serde_as_str)]
+        pub file: Option<PathBuf>
     }
 }
 
 impl Default for ConfigurationProxy {
     fn default() -> Self {
         Self {
-            account_public_key: None,
-            account_private_key: Some(None),
+            public_key: None,
+            private_key: Some(None),
+            file: None,
         }
     }
 }
 
+/// Parsed variant of the (user provided) [`Configuration`]
+// TODO: incorporate this struct into the final, parsed configuration
+//       https://github.com/hyperledger/iroha/issues/3500
+pub enum ParsedConfiguration {
+    /// The peer can only observe the genesis block
+    Default {
+        /// Genesis account public key
+        public_key: PublicKey,
+    },
+    /// The peer is responsible for submitting of the genesis block
+    Submit {
+        /// Genesis account key pair
+        key_pair: KeyPair,
+        /// Raw genesis block
+        raw_block: RawGenesisBlock,
+    },
```

Comment:
Better naming ideas?

What I have: `Partial`/`Full`, `PublicKey`/`Full`

Reply:
I think `Partial`/`Full` is the best one among the options you suggested.

maybe something like `ObserveGenesis`/`SubmitGenesis`? Or `View`/`Submit`?

Reply:
> maybe something like `ObserveGenesis`/`SubmitGenesis`? Or `View`/`Submit`?

I like these options too. Let's wait for more opinions.

Reply:
Since there are not many more opinions here, I'll use `Partial`/`Full`. Not necessarily in this PR, but as part of https://github.com/hyperledger/iroha/issues/4161

Reply:
Changed it now